parachain/runtime/kintsugi/src/dex.rs

```diff
@@ -0,0 +1,243 @@
+    if let Err(err) = DexGeneral::create_pair(RuntimeOrigin::root(), asset_0, asset_1, fee_rate) {
+        log::error!("Could not create pair: {:?}", err);
+    } else {
+        if let Err(err) = Farming::update_reward_schedule(
+            RuntimeOrigin::root(),
+            CurrencyId::join_lp_token(asset_0, asset_1).expect("currencies are valid; qed"),
+            CurrencyId::Token(KINT),
+            132480, // 60 * 24 * 92 (approx three months)
+            rewards,
+        ) {
+            log::error!("Could not update rewards: {:?}", err);
+        }
+    }
+}
+
+impl OnRuntimeUpgrade for DexSetup {
+    fn on_runtime_upgrade() -> Weight {
+        create_pair_and_update_rewards(
+            CurrencyId::Token(KBTC),
+            CurrencyId::Token(KSM),
+            45000000000000000, // 45,000 KINT
+            15,                // 0.15%
+        );
+
+        create_pair_and_update_rewards(
+            CurrencyId::Token(KBTC),
+            CurrencyId::ForeignAsset(3), // USDT
+            40000000000000000,           // 40,000 KINT
+            15,                          // 0.15%
+        );
+
+        create_pair_and_update_rewards(
+            CurrencyId::Token(KSM),
+            CurrencyId::Token(KINT),
+            35000000000000000, // 35,000 KINT
+            25,                // 0.25%
+        );
+
+        Default::default()
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+        ensure!(
+            dex_general::PairStatuses::<Runtime>::iter().collect::<Vec<_>>().len() == 0,
+            "Should not have pools"
+        );
+        Ok(Vec::new())
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn post_upgrade(_state: Vec<u8>) -> Result<(), &'static str> {
+        ensure!(
+            dex_general::PairStatuses::<Runtime>::iter().collect::<Vec<_>>().len() == 3,
+            "Should have pools"
+        );
+        Ok(())
+    }
+}
+
+pub struct LoansSetup;
+
+fn add_and_activate_market(token: CurrencyId, market: loans::Market<Balance>) {
+    if let Err(err) = Loans::add_market(RuntimeOrigin::root(), token, market) {
+        log::error!("Could not add market: {:?}", err);
+    } else {
+        if let Err(err) = Loans::activate_market(RuntimeOrigin::root(), token) {
+            log::error!("Could not activate market: {:?}", err);
+        }
+    }
+}
+
+impl OnRuntimeUpgrade for LoansSetup {
+    fn on_runtime_upgrade() -> Weight {
+        add_and_activate_market(
+            CurrencyId::Token(KBTC),
+            loans::Market {
+                close_factor: Ratio::from_percent(50),
+                collateral_factor: Ratio::from_percent(63),
+                liquidation_threshold: Ratio::from_percent(67),
+                liquidate_incentive: Rate::from_inner(Rate::DIV / 100 * 110),
+                liquidate_incentive_reserved_factor: Ratio::zero(),
```

Comment:
This should be 2.5% everywhere: https://www.notion.so/interlay/kBTC-9c73abf3829c49c684b344efec256032?pvs=4

Reply:
Not according to this: https://docs.google.com/spreadsheets/d/1U_iFh8Bx4pRNOkjYFEL0jwOlb1KqKgnjAJXkFCCx1M0/edit#gid=1950019562